File: loquacious/src/test/java/com/saantiaguilera/loquacious/model/ItemTest.java
package com.saantiaguilera.loquacious.model;

import org.junit.Assert;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.robolectric.RobolectricTestRunner;

/**
 * Created by saguilera on 11/18/17.
 */
@RunWith(RobolectricTestRunner.class)
public class ItemTest {

    @Test
    public void test_SingleItemHasMangledKey() {
        Item<String> item = new Item<>("test", "value");
        Assert.assertNotEquals("test", item.getKey());
        Assert.assertEquals("value", item.getValue());
    }

    @Test
    public void test_QuantityItemHasMangledKey() {
        Item<String> item = new Item<>("test", Quantity.ZERO, "value");
        Assert.assertNotEquals("test", item.getKey());
        Assert.assertNotEquals(Quantity.ZERO, item.getKey());
        Assert.assertEquals("value", item.getValue());
    }

}

File: loquacious/src/test/java/com/saantiaguilera/loquacious/model/ItemTest.kt
package com.saantiaguilera.loquacious.model

import org.junit.Assert
import org.junit.Test
import org.junit.runner.RunWith
import org.robolectric.RobolectricTestRunner

/**
 * Created by saguilera on 11/18/17.
 */
@RunWith(RobolectricTestRunner::class)
class ItemTest {

    @Test
    fun test_SingleItemHasMangledKey() {
        val item = Item("test", "value")
        Assert.assertNotEquals("test", item.key)
        Assert.assertEquals("value", item.value)
    }

    @Test
    fun test_QuantityItemHasMangledKey() {
        val item = Item("test", Quantity.ZERO, "value")
        Assert.assertNotEquals("test", item.key)
        Assert.assertNotEquals(Quantity.ZERO, item.key)
        Assert.assertEquals("value", item.value)
    }

}
